src/Dropdown.elm

```diff
@@ -35,44 +35,54 @@ type DropdownType
     | MultiSelect
 
 
-type alias InternalState item =
+type alias InternalState =
     { id : String
     , isOpen : Bool
-    , selectedItems : List item
     , filterText : String
     , focusedIndex : Int
     }
 
 
-{-| Type that holds the handlers for single and multiple selects. |
+{-| Types that hold the handlers for single and multiple selects. |
 -}
-type OnSelectMsg msg item
-    = SingleItem (item -> Element msg)
-    | MultipleSelection (List item -> Element msg)
+type SelectionToPrompt item msg
+    = SingleItemToPrompt (item -> Element msg)
+    | MultipleItemsToPrompt (List item -> Element msg)
+
+
+type OnSelectMsg item msg
+    = OnSelectSingleItem (Maybe item -> msg)
+    | OnSelectMultipleItems (List item -> msg)
+
+
+type Selection item
+    = SingleItem (Maybe item)
+    | MultipleItems (List item)
 
 
 {-| Opaque type that holds the current state
 
     type alias Model =
-        { dropdownState : Dropdown.State String
+        { dropdownState : Dropdown.State
         }
 
 -}
-type State item
-    = State (InternalState item)
+type State
```

Comment:
I removed the item parameter from the public State type as it's not needed in the context of this PR and my IDE was complaining about an unused parameter, but now I'm wondering if it would be better left in - to minimise the api changes in case in future a change is made to the state that requires an item e.g. refactoring focusedIndex to be an item instead of an index or something like that. What do people think? I'm happy to add this back in if preferred